psh/pshapp/env.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Phoenix-RTOS
+ *
+ * psh variables and environment
+ *
+ * implements psh builtins:
+ * - export: set and propagate variable names to environment
+ * - unset:  removes variables whose names match
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#include "../psh.h"
+
+
+extern char **environ;
+
+
+static int psh_validateVarName(const char *name, int len)
+{
+	if ((name == NULL) || (*name == '\0') || (isdigit(*name) != 0)) {
```

Comment:
Shouldn't we require a letter instead?
```suggestion
	if ((name == NULL) || (*name == '\0') || (isalpha(*name) == 0)) {
```

Reply:
Right, but in sense `... || (isdigit(*name) != 0)  || (isalpha(*name) == 0)`. Digits at first character position are forbidden.